mlflow/server/handlers.py

```diff
@@ -1363,6 +1364,25 @@ def _delete_registered_model_tag():
     return _wrap_response(DeleteRegisteredModelTag.Response())
 
 
+def _validate_source(source: str, run_id: str) -> None:
+    if not is_local_uri(source):
+        return
+
+    if run_id:
+        store = _get_tracking_store()
+        run = store.get_run(run_id)
+        source = pathlib.Path(local_file_uri_to_path(source))
+        run_artifact_dir = pathlib.Path(local_file_uri_to_path(run.info.artifact_uri))
+        if run_artifact_dir in [source, *source.parents]:
+            return
```

Comment:
Can we add some unit tests for this part ?
for both UNIX and Windows.

Reply:
tests/tracking/test_rest_tracking.py runs both on Unix and Windows.